Sensor

asadate opened the pull request:
Updated sensors specifications.

asadate committed:
Update definitions.json

asadate committed:
Create sensor.json

asadate committed:
Update sensor.json

Added additional properties for the accelerometer and the gyro sensors

rsnancollas commented on part-spec/definitions.json:
```diff
@@ -455,6 +455,26 @@
+      "properties": {
+        "minValue":{
+          "description": "minimum value of a component",
+          "type": "number",
```
To be more similar to how we've defined this stuff in other parts of the spec:

remove
"type": "number"

asadate replied:
fixed

rsnancollas commented on part-spec/definitions.json:
```diff
@@ -455,6 +455,26 @@
+        },
+        "typValue":{
+          "description": "typical value of a component",
+          "type": "number",
```
same as above

asadate replied:
fixed

rsnancollas commented on part-spec/definitions.json:
```diff
@@ -455,6 +455,26 @@
+        },
+        "maxValue":{
+          "description": "maximum value of a component",
+          "type": "number",
```
same as above

asadate replied:
fixed

rsnancollas commented on part-spec/sensor.json:
```diff
@@ -0,0 +1,240 @@
+                    "$ref": "definitions.json#/componentID"
+                },
+                "accelerometerType": {
+                    "description": "Type of accelerometer",
```
suggest being consistent in casing:

type of accelerometer

asadate replied:
fixed

rsnancollas commented on part-spec/sensor.json:
```diff
@@ -0,0 +1,240 @@
+                    "type": "string"
+                },
+                "accelerationRanges": {
+                    "description": "range of force that accelerometer can measure",
```
this is more like the maximum force it can measure, right?

Carrotman42 replied:
Yeah, if we need a range we will need a min and a max.

asadate replied:
Yes, Maximum force but some parts have this as a programmable value with different ranges. Could change as an array. Let's discuss more.

rsnancollas commented on part-spec/sensor.json:
```diff
@@ -0,0 +1,240 @@
+                    "type": "number"
+                },
+                "accelerationSensitivity": {
+                    "description": "smallest change in force the accelerometer is able to capture (typical). Corresponds to levels in acceleration ranges",
```
"description": "smallest change in force the accelerometer is able to capture (typical) at a given acceleration range",

asadate replied:
fixed

rsnancollas commented on part-spec/sensor.json:
```diff
@@ -0,0 +1,240 @@
+                        "2000"
+                    ],
+                    "comment": "units of LSB/g; conditions can include acceleration ranges and output resolution",
+                    "type": "number",
```
remove the type: number, the type is covered by the "$ref"

asadate replied:
done

rsnancollas commented on part-spec/sensor.json:
```diff
@@ -0,0 +1,240 @@
+                "accelerationSensitivityOverTemperature": {
+                    "description": "accelerometer sensitivity change over temperature",
+                    "comment": "units of percentage per degree celsius (%/Celcius)",
+                    "$ref": "definitions.json#/value"
```
I don't think %/deg Celsius is a value that we define, so this should be:

"type": "number"

asadate replied:
fixed

rsnancollas replied:
Looks like this didn't get picked up, I think we want:

"type: "number"

NOT: "$ref": "definitions.json#/value"

rsnancollas commented on part-spec/sensor.json:
```diff
@@ -0,0 +1,240 @@
+                },
+                "accelerationSensitivityOverTemperature": {
+                    "description": "accelerometer sensitivity change over temperature",
+                    "comment": "units of percentage per degree celsius (%/Celcius)",
```
Celsius

asadate replied:
fixed

rsnancollas commented on part-spec/sensor.json:
```diff
@@ -0,0 +1,240 @@
+                "zerogOffset": {
+                    "description": "output of the accelerometer when no acceleration is applied",
+                    "comment": "units of g-force (g)",
+                    "$ref": "definitions.json#/value"
```
Again, we are not defining g-force in value.

We can either add it or just list this as a number

asadate replied:
number is OK since I don't think this unit gets used that much outside of this

rsnancollas replied:
Looks like this didn't get picked up, I think we want:

"type: "number"

NOT: "$ref": "definitions.json#/value"

rsnancollas commented on part-spec/sensor.json:
```diff
@@ -0,0 +1,240 @@
+                "zerogOffsetOverTemperature": {
+                    "description": "accelerometer zero-g offset change over temperature",
+                    "comment": "units of milli g-force per degree celsius (mg/Celcius)",
+                    "$ref": "definitions.json#/value"
```
Number, not "$ref"

asadate replied:
done

rsnancollas replied:
Looks like this didn't get picked up, I think we want:

"type: "number"

NOT: "$ref": "definitions.json#/value"

asadate replied:
"$ref": "definitions.json#/value" should be ok here since we've redefined this as a range of numbers ( min, typ, max). This is for range values without defined units ( in the definitions file).

rsnancollas commented on part-spec/sensor.json:
```diff
@@ -0,0 +1,240 @@
+                },
+                "outputDataRate": {
+                    "description": "Output Data rate (ODR) of a device",
+                    "comment": "units of Hz",
```
Hz isn't really a data rate. I think we want bps

asadate replied:
Datasheets are using Hz

rsnancollas commented on part-spec/sensor.json:
```diff
@@ -0,0 +1,240 @@
+                "rmsNoise": {
+                    "description": "Broadband rms noise of a device",
+                    "comment": "units of milli g-force",
+                    "$ref": "definitions.json#/value"
```
number, not "$ref" unless we add this value

asadate replied:
done

rsnancollas replied:
Looks like this didn't get picked up, I think we want:

"type: "number"

NOT: "$ref": "definitions.json#/value"

asadate replied:
Same comment as above.

rsnancollas commented on part-spec/sensor.json:
```diff
@@ -0,0 +1,240 @@
+                "spectralNoiseDensity": {
+                    "description": "Spectral noise density of a device",
+                    "comment": "units of milli g-force per square root of Hz (milli-g/rt-Hz)",
+                    "$ref": "definitions.json#/value"
```
again, just a number

asadate replied:
done

rsnancollas replied:
Looks like this didn't get picked up, I think we want:

"type: "number"

NOT: "$ref": "definitions.json#/value"

rsnancollas commented on part-spec/sensor.json:
```diff
@@ -0,0 +1,240 @@
+                },
+                "gyroRanges": {
+                    "description": "range of angular speed that gyro can measure",
+                    "examples": [
```
I'm not sure these examples are that useful since they're just numbers

asadate replied:
removed

rsnancollas commented on part-spec/sensor.json:
```diff
@@ -0,0 +1,240 @@
+                },
+                "gyroSensitivity": {
+                    "description": "smallest change in angular speed gyro is able to capture (typical). Corresponds to levels in gyro ranges",
+                    "examples": [
```
Again, not sure the examples do that much

rsnancollas replied:
Also the examples should be conditional Properties, so I would remove them here

asadate replied:
removed

rsnancollas commented on part-spec/sensor.json:
```diff
@@ -0,0 +1,240 @@
+                        "65"
+                    ],
+                    "comment": "units of LSB/dps",
+                    "type": "number",
```
remove type: number, the type is the conditionalProperty

asadate replied:
removed

rsnancollas commented on part-spec/sensor.json:
```diff
@@ -0,0 +1,240 @@
+                "gyroSensitivityOverTemperature": {
+                    "description": "gyroscope sensitivity change over temperature",
+                    "comment": "units of percentage per degree celsius",
+                    "$ref": "definitions.json#/value"
```
number, not value

asadate replied:
done

rsnancollas replied:
Looks like this didn't get picked up, I think we want:

"type: "number"

NOT: "$ref": "definitions.json#/value"

rsnancollas commented on part-spec/sensor.json:
```diff
@@ -0,0 +1,240 @@
+                "zeroRateOffset": {
+                    "description": "output of the gyroscope when no angular velocity is applied",
+                    "comment": "units of degrees-per-second",
+                    "$ref": "definitions.json#/value"
```
number, not value

asadate replied:
fixed

Carrotman42 replied:
looks like this change is not included in the latest upload?

rsnancollas replied:
Looks like this didn't get picked up, I think we want:

"type: "number"

NOT: "$ref": "definitions.json#/value"

asadate replied:
Number is already the defined type from Value object.

rsnancollas commented on part-spec/sensor.json:
```diff
@@ -0,0 +1,240 @@
+                "zeroRateOffsetOverTemperature": {
+                    "description": "gyro zero rate offset change over temperature",
+                    "comment": "units of degrees-per-second per degree celsius (dps/Celcius)",
+                    "$ref": "definitions.json#/value"
```
number, not value

asadate replied:
fixed

rsnancollas replied:
Looks like this didn't get picked up, I think we want:

"type: "number"

NOT: "$ref": "definitions.json#/value"

asadate replied:
Number is already the defined type from the value object.

rsnancollas commented on part-spec/sensor.json:
```diff
@@ -0,0 +1,240 @@
+                },
+                "outputDataRate": {
+                    "description": "Output Data rate (ODR) of a device",
+                    "comment": "units of Hz",
```
Hz or bps?

asadate replied:
Shown as Hz in datasheets

rsnancollas commented on part-spec/sensor.json:
```diff
@@ -0,0 +1,240 @@
+                "rmsNoise": {
+                    "description": "Broadband rms noise of a device",
+                    "comment": "units of degrees-per-second rms",
+                    "$ref": "definitions.json#/value"
```
number, not value

asadate replied:
done

rsnancollas replied:
Looks like this didn't get picked up, I think we want:

"type: "number"

NOT: "$ref": "definitions.json#/value"

rsnancollas commented on part-spec/sensor.json:
```diff
@@ -0,0 +1,240 @@
+                "spectralNoiseDensity": {
+                    "description": "Spectral noise density of a device",
+                    "comment": "units of degrees-per-second per square root Hertz (dps/rt-Hz)",
+                    "$ref": "definitions.json#/value"
```
number, not value

asadate replied:
done

rsnancollas replied:
Looks like this didn't get picked up, I think we want:

"type: "number"

NOT: "$ref": "definitions.json#/value"

rsnancollas commented on part-spec/sensor.json:
```diff
@@ -0,0 +1,240 @@
+                    "type": "string"
+                },
+                "rmsNoise": {
+                    "description": "Broadband rms noise of a device",
```
lowercase Broadband

asadate replied:
done

rsnancollas commented on part-spec/sensor.json:
```diff
@@ -0,0 +1,240 @@
+                    "$ref": "definitions.json#/value"
+                },
+                "spectralNoiseDensity": {
+                    "description": "Spectral noise density of a device",
```
lowercase Spectral

asadate replied:
done

rsnancollas reviewed (commented):
The main type of comment I had was making sure that if we reference "$ref": "#/definitions/unit", the unit definition actually supports the unit we're referencing. Options are either to add that type of unit to the definition or just use type: number with a comment about what the unit is.

Carrotman42 commented on part-spec/sensor.json:
```diff
@@ -0,0 +1,240 @@
+                        "4000",
+                        "2000"
+                    ],
+                    "comment": "units of LSB/g; conditions can include acceleration ranges and output resolution",
```
I see LSB/g is a standard thing (I found the definition as the first result on Google), but maybe we want to define it here so that our spec can be more standalone. e.g. I'm thinking just:

"units of LSB/g   (least significant bits per g)

asadate replied:
fixed

Carrotman42 commented on part-spec/sensor.json:
```diff
@@ -0,0 +1,240 @@
+                    "type": "number"
+                },
+                "interface": {
+                    "description": "interface(s) for communication to host",
```
If we want to support multiple interfaces, we should take this as an array of strings rather than as a stirng.

asadate replied:
changed to array

Carrotman42 replied:
With an array type we need to specify the type of the inner element, and it looks like we do that via the "items" field: https://github.com/edatasheets/digital-datasheets/blob/main/part-spec/power.json#L133

Please add that field here and throughout where relevant.

asadate replied:
changed

Carrotman42 commented on part-spec/sensor.json:
```diff
@@ -0,0 +1,240 @@
+                },
+                "currentConsumption": {
+                    "description": "current consumption of a device",
+                    "type": "array",
```
I'm a little confused why this is an array - why are there multiple different currents, and how will consumers of the datasheet know which one to look at?

asadate replied:
Probably got confused with my types again. Let me think more about this. It's in the definition file.

asadate replied:
changed

Carrotman42 commented on part-spec/sensor.json:
```diff
@@ -0,0 +1,240 @@
+            }
+        },
+        {
+            "title": "gyro",
```
should we use the full name here? i.e. gyroscope

I don't feel strongly, but I don't see a reason to use colloquialisms.

asadate replied:
Good point. Changed.

Carrotman42 commented on part-spec/sensor.json:
```diff
@@ -0,0 +1,240 @@
+                },
+                "zeroRateOffsetOverTemperature": {
+                    "description": "gyro zero rate offset change over temperature",
+                    "comment": "units of degrees-per-second per degree celsius (dps/Celcius)",
```
I'm probably confused: `degrees-per-second per degree celsius` sounds like `celsius-per-second per celsius` which would be `C/s / C` => `1/s` => `Hz`.

Unless this is somehow very obvious to folks familiar with gyroscopes, can we update this comment to be more clear? (I don't really have a great suggestion because I think I just don't understand this :] )

asadate replied:
degrees (angle) vs degrees ( temperature). But yes, it can be confusing.

Carrotman42 replied:
oh I see, thanks!

Carrotman42 commented on part-spec/sensor.json:
```diff
@@ -0,0 +1,240 @@
+                        "spi",
+                        "analog"
+                    ],
+                    "type": "string"
```
Ditto here about making tihs an array if we expect multiple interfaces

asadate replied:
changed to array

asadate committed:
Update definitions.json

asadate committed:
Update sensor.json

asadate committed:
Update sensor.json

asadate committed:
Update definitions.json

Added typValue/minValue/maxValue to the units object

asadate committed:
Update sensor.json

asadate committed:
Update definitions.json

updated currentConsumption Object - Remove typ/min/max since they're now included as part of unit object

asadate committed:
Update definitions.json

asadate committed:
Update sensor.json

asadate committed:
Update sensor.json

Added items for the array types

Carrotman42 commented on part-spec/sensor.json:
```diff
@@ -171,7 +174,10 @@
-                    "type": "array"
+                    "type": "array",
+                    "items: {
+                        "type": "string
```
Please fix missing end quote here: should be `"string"`

Carrotman42 reviewed (approved):
I will preemptively approve to unblock you, but as per the email thread: please make sure the changes Rachel requested make it up to github before submitting.

asadate committed:
Update sensor.json
